Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/StartRide.hs

```diff
@@ -152,25 +153,26 @@ startRide ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.getId)
   unless (isValidRideStatus (ride.status)) $ throwError $ RideInvalidStatus "This ride cannot be started"
 
   enableLocationTrackingService <- asks (.enableLocationTrackingService)
-  point <- case req of
+  (point, odometerStartReading) <- case req of
     DriverReq driverReq -> do
       when (driverReq.rideOtp /= ride.otp) $ throwError IncorrectOTP
+      when (booking.bookingType == SRB.RentalBooking && isNothing driverReq.odometerStartReading) $ throwError InvalidRideRequest
```

Comment:
Error should be E400 and also throw proper context in error message saying please provide proper odometerStartReading or something.

Reply:
Also, shouldn't we validate this value with customer once?